### PR TITLE
Remove `is_canvas` from the `LTILaunchResource`

### PR DESCRIPTION
**Pull request by jon-betts**
Requires:

 * https://github.com/hypothesis/lms/pull/5224

I had originally thought to try removing all parts, but I think the next two are a bit harder for different reasons.

There should be no functional differences here.

### CODE REVIEW

File: lms/resources/lti_launch.py [hunk 1]
```diff
@@ -3,23 +3,13 @@
...
-    """
-    Context resource for LTI launch requests.
-
-    Many methods and properties of this class are only meant to be called when
-    request.parsed_params holds validated params from an LTI launch request and
-    might crash otherwise. So you should only call these methods after the
-    request has been validated with BasicLTILaunchSchema or similar (for
-    example from views that have schema=BasicLTILaunchSchema in their view
-    config).
-    """
```
Comment (jon-betts) [inline lms/resources/lti_launch.py:22]: I don't think this is really true / relevant any more.

File: tests/unit/lms/resources/_js_config/__init___test.py [hunk 2]
```diff
@@ -646,7 +646,6 @@ def context(application_instance):
         LTILaunchResource,
         spec_set=True,
         instance=True,
-        is_canvas=True,
```
Comment (jon-betts) [inline tests/unit/lms/resources/_js_config/__init___test.py:649]: I think this is a left over from the previous changes.

File: tests/unit/lms/resources/_js_config/file_picker_config_test.py [hunk 3]
```diff
@@ -149,13 +149,13 @@ def test_jstor_config(self, pyramid_request, jstor_service, enabled):
...
     @pytest.fixture
-    @pytest.mark.usefixtures("with_is_canvas")
+    @pytest.mark.usefixtures("with_canvas")
     def canvas_files_enabled(self, pyramid_request, application_instance):
         pyramid_request.params["custom_canvas_course_id"] = sentinel.course_id
         application_instance.developer_key = sentinel.developer_key
 
     @pytest.fixture
-    def with_is_canvas(self, pyramid_request):
+    def with_canvas(self, pyramid_request):
```
Comment (jon-betts) [inline tests/unit/lms/resources/_js_config/file_picker_config_test.py:158]: This is just a weird name now, and crops up when grepping.

File: tests/unit/lms/resources/lti_launch_test.py [hunk 4]
```diff
@@ -1,37 +1,18 @@
...
-        [
-            (Product.Family.CANVAS, True),
-            (Product.Family.BLACKBOARD, False),
-            (Product.Family.UNKNOWN, False),
-        ],
-    )
-    def test_is_canvas(self, pyramid_request, product, expected):
-        pyramid_request.product.family = product
-
-        assert LTILaunchResource(pyramid_request).is_canvas == expected
```
Comment (jon-betts) [inline tests/unit/lms/resources/lti_launch_test.py:33]: We now run the risk that if statements in Python no longer work.

**Review by marcospri — APPROVED** [review]
Yup, this looks like satisfying one :+1:

**Pull request merged**